### PR TITLE
Sequelize Setup

### PR DESCRIPTION
**Pull request by allopez7**
* Setup sequelize with testing
* Added npm start which gives the ability to run the project

### CODE REVIEW

File: sequelize/dbQueries.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const {Account} = require('./db')
```
Comment (allopez7) [inline sequelize/dbQueries.js:1]: Did not intend to push this file

File: index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require('dotenv').config()
```
Comment (songz) [inline index.js:1]: remove this file, not needed in the PR.

File: sequelize/db.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+const { Sequelize, DataTypes } = require('sequelize')
+
+let sequelize
+const dbModule = {}
+
+dbModule.start = ()=>{
+  sequelize = new Sequelize(`${process.env.DATABASE}`, `${process.env.USERNAME}`, `${process.env.PASSWORD}`, {
```
Comment (songz) [inline sequelize/db.js:7]: these are already strings, no need for backticks.

File: sequelize/db.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    host: process.env.HOST || 'localhost', 
+    dialect: process.env.DIALECT || 'postgres',
```
Comment (songz) [inline sequelize/db.js:9]: `dialect` could be hardcoded to postgres.

File: package.json [hunk 5]
```diff
@@ -4,6 +4,7 @@
   "description": "Create databases in postgres, mongoDB and neo4J",
   "main": "index.js",
   "scripts": {
+    "start": "node index.js",
```
Comment (songz) [inline package.json:7]: not in this PR

File: package.json [hunk 6]
```diff
@@ -19,11 +20,13 @@
   "devDependencies": {
     "dotenv": "^8.2.0",
     "husky": "^4.2.3",
-    "jest": "^25.2.4",
+    "jest": "^25.3.0",
     "pm2": "^4.2.3"
   },
   "dependencies": {
-    "pg": "^8.0.0",
+    "express": "^4.17.1",
```
Comment (songz) [inline package.json:27]: not in this PR

File: sequelize/db.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      max: 5,
+      min: 0,
+      idle: 10000
+    }
+  })
+  return sequelize.authenticate()
+}
+
+dbModule.Account = ()=>{
+  return sequelize.define('account', {
```
Comment (songz) [inline sequelize/db.js:20]: We will not be defining Account other than the time when sequelize has started, so this function is not necessary. Should be added to `start` function.

File: sequelize/db.test.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    close()
+    expect(mockSequelize.close).toHaveBeenCalledTimes(1)
+  })
+  it('should test throw on sequelize authentication', async ()=>{
+    try{
+      const resStart = await start()
+      await mockSequelize.authenticate.mockReturnValue(Promise.reject()) 
+      expect(resStart).rejects.toThrow()
+    }catch(err){
+      expect(1).toHaveBeenCalledTimes(1)
```
Comment (allopez7) [inline sequelize/db.test.js:38]: I had logged here using `logger.error` after creating a `logger = { error: jest.fn() }` but I got zero times called.

File: sequelize/db.test.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
+jest.mock('sequelize')
+const {Sequelize} = require('sequelize')
+const {start, update, close, Account} = require('./db')
+
```
Comment (allopez7) [inline sequelize/db.test.js:4]: had `require('winston')` and `jest.mock('winston')` set up here causing the error i sent.

File: sequelize/db.test.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    expect(mockSequelize.close).toHaveBeenCalledTimes(1)
+  })
+  it('should test throw on sequelize authentication', async ()=>{
+    try{
+      const resStart = await start()
+      await mockSequelize.authenticate.mockReturnValue(Promise.reject()) 
+      expect(resStart).rejects.toThrow()
+    }catch(err){
+      const logger = require('../lib/log')(__filename)
+      expect(logger.error).toHaveBeenCalledTimes(1)
```
Comment (allopez7) [inline sequelize/db.test.js:40]: i know this is wrong, i saw the result and the logger.error is a function.

File: sequelize/db.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
+const { Sequelize, DataTypes } = require('sequelize')
+const logger = require('../lib/log')(__filename)
+require('dotenv').config()
+
+let sequelize
+const dbModule = {}
+
+dbModule.start = async ()=>{
```
Comment (songz) [inline sequelize/db.js:8]: `logger.log('starting sequelize server')`

File: sequelize/db.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+      emailConfirmation: {
+        type: DataTypes.STRING
+      },
+      passwordReset: {
+        type: DataTypes.STRING
+      },
+      databases: {
+        type: DataTypes.TEXT
+      }
+    })
```
Comment (songz) [inline sequelize/db.js:46]: `logger.log('sequelize authenticating...')`

File: sequelize/db.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+    await sequelize.authenticate()
+  }catch(err){
+    logger.error(err)
+    throw new Error('failed to authenticate sequelize account')
+  }
+}
+
+dbModule.update = async ()=>{
+  return sequelize.sync({alter: true})
+}
```
Comment (songz) [inline sequelize/db.js:56]: I think its better to toggle this using an env variable and then directly run `sync` in the `start` call. So delete this function and after authenticate do this:

```
await sequelize.sync( { alter: !!process.env.ALTER_DB })
```

**Pull request merged**